be/src/common/config_update_func.h

```diff
@@ -0,0 +1,89 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef DORIS_BE_SRC_COMMON_CONFIG_UPDATE_FUNC_H
+#define DORIS_BE_SRC_COMMON_CONFIG_UPDATE_FUNC_H
+
+#include "common/status.h"
+#include "configbase.h"
+#include "common/config.h"
+#include "gutil/strings/substitute.h"
+
+namespace doris {
+class Status;
+
+namespace config {
+
+/// regist update check function of config `cumulative_size_based_promotion_size_mbytes`
+/// first, check `cumulative_size_based_promotion_size_mbytes` is greater than or equals 
+/// `cumulative_size_based_promotion_min_size_mbytes`
+/// second, check `cumulative_size_based_promotion_size_mbytes` is greater than or equals 
+/// 2 * `cumulative_size_based_compaction_lower_size_mbytes`
+CONF_Update_Check(cumulative_size_based_promotion_size_mbytes, [](void* config) {
```

Comment:
Have these configs been checked when read from be.conf?
How about register these check functions when defined them in config.h? It will be closer to the place where we defined them, and avoid using a mistake config name, for example `cumulative_size_based_promotion_size_mbyte `.

Reply:
Configs have been checked.